cabot/statuscheckapp/models/base.py

```diff
@@ -0,0 +1,137 @@
+from celery.utils.log import get_task_logger
+from django.db import models
+import json
+import time
+from cabot.cabotapp.models import CHECK_TYPES, StatusCheck, StatusCheckResult
+
+logger = get_task_logger(__name__)
+
+
+class MetricsInstanceBase(models.Model):
+    class Meta:
+        app_label = 'statuscheckapp'
+
+    name = models.CharField(
+        unique=True,
+        max_length=30,
+        help_text='Unique name for the data source',
+    )
+
+
+class MetricsStatusCheckBase(StatusCheck):
+    class Meta:
+        app_label = 'statuscheckapp'
+
+    metrics_instance = models.ForeignKey(MetricsInstanceBase)
+    check_type = models.CharField(
+        choices=CHECK_TYPES,
+        max_length=30,
+    )
+    value = models.FloatField(
+        help_text='The check fails if this expression evaluates to False.',
+    )
+    time_range = models.IntegerField(
+        default=5,
+        help_text='Time range the check gathers data for.',
+    )
+
+    def _get_series(self):
+        """
+        Get the data series for this check and parse into a generic format.
+        Implemented by subclasses.
+        :return: the data series in the format
```

Comment:
Sure, that makes sense. Should I switch to yaml from json? Would there be any reason either one would be better?